### PR TITLE
Mpuncel/farm error reporting

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: pkg/rc/farm.go [hunk 1]
```diff
@@ -65,22 +67,27 @@ func NewFarm(
 	logger logging.Logger,
 	rcSelector klabels.Selector,
 	alerter alerting.Alerter,
+	errorReporter error_reporter.Reporter,
```
Comment (mpuncel) [inline pkg/rc/farm.go]: alternatively I could make the constructor place a no-op reporter on the struct and then add a SetErrorReporter function. that might be a little nicer than spreading `nil` everywhere

Comment (rudle) [inline pkg/rc/farm.go]: I don't see any precedent for that style of injection. I don't mind the extra nil in constructors fwiw

**Comment by rudle** [comment]
👍

File: pkg/rc/farm.go (hunk 1 above)
Comment (mpuncel) [inline pkg/rc/farm.go]: i did do that for metrics registry in the farms I think, but i also don't mind the `nil`

**Pull request merged**